src/Microsoft.Diagnostics.Monitoring/DiagnosticsMonitor.cs

```diff
@@ -0,0 +1,377 @@
+            {
+                // Level, FactoryID, LoggerName, EventID, EventName, ExceptionJson, ArgumentsJson
+                var logLevel = (LogLevel)traceEvent.PayloadByName("Level");
+                var factoryId = (int)traceEvent.PayloadByName("FactoryID");
+                var categoryName = (string)traceEvent.PayloadByName("LoggerName");
+                var eventId = (int)traceEvent.PayloadByName("EventId");
+                var eventName = (string)traceEvent.PayloadByName("EventName");
+                var exceptionJson = (string)traceEvent.PayloadByName("ExceptionJson");
+                var argsJson = (string)traceEvent.PayloadByName("ArgumentsJson");
+
+                // There's a bug that causes some of the columns to get mixed up
+                if (eventName.StartsWith("{"))
+                {
+                    argsJson = exceptionJson;
+                    exceptionJson = eventName;
+                    eventName = null;
+                }
+
+                if (string.IsNullOrEmpty(argsJson))
+                {
+                    return;
+                }
+
+                Exception exception = null;
+
+                ILogger logger = _services.GetService<ILoggerFactory>().CreateLogger(categoryName);
+
+                var scopes = new List<IDisposable>();
+
+                if (logActivities.TryGetValue(traceEvent.ActivityID, out var logActivityItem))
+                {
+                    // REVIEW: Does order matter here? We're combining everything anyways.
+                    while (logActivityItem != null)
+                    {
+                        scopes.Add(logger.BeginScope(logActivityItem.ScopedObject));
+
+                        logActivityItem = logActivityItem.Parent;
+                    }
+                }
+
+                try
+                {
+                    if (exceptionJson != "{}")
+                    {
+                        var exceptionMessage = JsonSerializer.Deserialize<JsonElement>(exceptionJson);
+                        exception = new LoggerException(exceptionMessage);
+                    }
+
+                    var message = JsonSerializer.Deserialize<JsonElement>(argsJson);
+                    if (message.TryGetProperty("{OriginalFormat}", out var formatElement))
+                    {
+                        var formatString = formatElement.GetString();
+                        var formatter = new LogValuesFormatter(formatString);
+                        object[] args = new object[formatter.ValueNames.Count];
+                        for (int i = 0; i < args.Length; i++)
+                        {
+                            args[i] = message.GetProperty(formatter.ValueNames[i]).GetString();
+                        }
+
+                        logger.Log(logLevel, new EventId(eventId, eventName), exception, formatString, args);
```

Comment:
I can think of two ways of going about this:
- Extend the ILogger interface to Log with timestamp. We can retain other potential log sinks by respecting both implementations, and passing along the timing data if the extended interface is implemented.
OR 
- Duplicate FormattedLogValues and attach a timestamp to it.
- Attach a timestamp to LogObject.

I'm leaning towards the first approach.

Reply:
By extend ILogger you mean derive a new interface such as ILoggerEx? And then we fork the code of ConsoleLogger into the monitor tool so that it implements ILoggerEx?

That sounds fine to me, but it also suggests that we shouldn't expect drop-in reusability of ILogger sinks. We have violated several assumptions that ILogger interface/sinks may make use of:
a) Log sink is invoked sufficiently close in time to the original ILogger.Log call that the timestamps can be considered indistinguishable
b) Log sink has access to original ILogger.Log() types (TState and Exception) and delegates (formatter). In our scenario the serialization that happens in EventSourceLogger could easily be lossy.
c) Log sink runs on same thread/same process as the call to ILogger.Log, implying it is fine to enrich the message with relevant process/thread specific information

My guess is that we could expect LoggerProviders to be repurposable with some work, but it is unlikely we can reference any of them as a drop-in NuGet package plugin and get a high quality results.